src/utils/uid.js

```diff
@@ -1,14 +1,16 @@
 
 import generate from 'uid'
 
+let N = 0
+
 /**
  * Create a unique identifier.
  *
  * @return {String} uid
  */
 
 function uid() {
-  return generate(4)
+  return (N++) + generate(4)
```

Comment:
We noticed a lot of conflicts in keys when the document is large; for example with a document containing 2000/3000 nodes, we were getting key conflict each time.

It reduces the probability of conflicts.